producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/ExecutionProducerRepositoryImpl.scala

```diff
@@ -90,27 +90,59 @@ class ExecutionProducerRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) exte
       Map("keys" -> events.map(_.planId))
     )
 
-    val progressNodes = events.map(e => Progress(
+    val queryPerformanceData = db.queryAs[Map[String, Any]](
```

Comment:
When naming a `Future` variables please use one of the following pattern (depending on what reads better in the context):
- `xxxFuture`
- `eventualXxx`
Same as you would do for `Option` identifiers, e.g. `maybeXxx` or `xxxOption`

Reply:
Also the phrase _performance data_ reads to me as the data related to the performance, i.e. some stats, metrics etc.
I would rather call it `execPlanDetails` or alike.